src/frontend/src/components/Header/Login.jsx

```diff
@@ -41,7 +34,7 @@ function Login() {
     getUserInfo();
   }, [telescopeUrl]);
 
-  return email ? <LoggedIn name={name} /> : <LoggedOut />;
+  return email ? { name } : false;
```

Comment:
I wonder if we can do something else with this function.  You're calling it like a function, but it's written like a component, with a React lifecycle and hooks.

There's a new API in React for doing what you want to do here, called [Context](https://reactjs.org/docs/context.html).  We need a way to globally manage this `user` info across all of our components, and broadcast changes.  We could pass it down from `PageBase` all the way to every other component via props.  But I think Context might be a better fit.

Let's talk tomorrow on Slack to figure out how to move forward with this.  I don't want to expand the scope so much that we can't land this in time for 1.0.

Reply:
Let's discuss this on Slack soon, when everyone is around, and make a plan.  @c3ho is suggesting to we try and do the Context piece in this PR.  If we need to do that, some of us could help make it happen.